README.md

```diff
@@ -86,24 +103,47 @@ You can use it like this: `argocd-vault-plugin generate /some/path -s vault-conf
 
 Note that this requires the `argocd-repo-server` to have a service account token mounted in the standard location.
 
-#### File on disk
+#### Configuration File
 The configuration can be given in a file reachable from the plugin, in any Viper supported format (YAML, JSON, etc.):
 ```yaml
-  VAULT_ADDR: Zm9v
-  AUTH_TYPE: Zm9v
-  GITHUB_TOKEN: Zm9v
-  TYPE: Zm9v
+VAULT_ADDR: Zm9v
+AUTH_TYPE: Zm9v
+GITHUB_TOKEN: Zm9v
+TYPE: Zm9v
 ```
 You can use it like this: `argocd-vault-plugin generate /some/path -c /path/to/config/file.yaml`. This can be useful for usecases not involving Argo CD.
 
-#### Environment variables
+#### Environment Variables
 The configuration can be set via environment variables, where each key is prefixed by `AVP_`:
 ```
 AVP_TYPE=vault # corresponds to TYPE key
 ```
 Make sure that these environment variables are available to the plugin when running it, whether that is in Argo CD or as a CLI tool. Note that any _set_
 environment variables take precedence over configuration pulled from a Kubernetes Secret or a file.
 
+### Full List of Supported Parameters
+
+| Name            | Description                | Notes                |
+| --------------- | -------------------------- | -------------------- |
+| VAULT_ADDR     | Address of your Vault      | N/A                  |
+| PATH_PREFIX    | Prefix of the vault path to look for the secrets | A `/` delimitted path to a secret in Vault. This value is concatenated with the `kind` of the given resource; e.g, replacing a Secret with `PATH_PREFIX` `my-team/my-app` will use the path `my-team/my-app/secret`       |
+| TYPE           | The type of Vault backend  | Supported values: `vault` and `secretmanager` |
+| KV_VERSION    | The vault secret engine  | Supported values: `1` and `2` (defaults to 2)|
```

Comment:
Might be good to explain the same for KV_VERSION, for when `kv_version` annotation is provided